cache/manager_test.go

```diff
@@ -1483,3 +1552,64 @@ func fileToBlob(file *os.File, compress bool) ([]byte, ocispecs.Descriptor, erro
 		},
 	}, nil
 }
+
+func mapToSystemTarBlob(m map[string]string) ([]byte, ocispecs.Descriptor, error) {
+	tmpdir, err := ioutil.TempDir("", "tarcreation")
+	if err != nil {
+		return nil, ocispecs.Descriptor{}, err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	expected := map[string]string{}
+	for k, v := range m {
+		expected[k] = v
+		if err := ioutil.WriteFile(filepath.Join(tmpdir, k), []byte(v), 0600); err != nil {
+			return nil, ocispecs.Descriptor{}, err
+		}
+	}
+
+	cmd := exec.Command("tar", "-C", tmpdir, "-c", ".")
```

Comment:
> But try it with a tarball from GNU tar for example. Tar allows padding etc so there are many ways how the same files could end up encoded into a tarball.

Conversion preserves the original tar bytes even for tar blobs created outside golang.